controllers/datafoundation.go

```diff
@@ -114,3 +148,61 @@ func (r *dataFoundationReconciler) reconcileOnboardingValidationSecret() error {
 	}
 	return nil
 }
+
+func (r *dataFoundationReconciler) reconcileCSV() error {
+	r.Log.Info("Reconciling OCS Operator CSV")
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err := r.list(&csvList); err != nil {
+		return fmt.Errorf("unable to list csv resources: %v", err)
+	}
+
+	for index := range csvList.Items {
+		csv := &csvList.Items[index]
+		if strings.HasPrefix(csv.Name, ocsOperatorName) {
+			if err := r.updateOCSCSV(csv); err != nil {
+				return fmt.Errorf("failed to update OCS CSV: %v", err)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func (r *dataFoundationReconciler) updateOCSCSV(csv *opv1a1.ClusterServiceVersion) error {
+	isChanged := false
+	r.addOwnerAnnotation(csv)
+	deployments := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+	for i := range deployments {
+		containers := deployments[i].Spec.Template.Spec.Containers
+		for j := range containers {
+			switch container := &containers[j]; container.Name {
+			case "ocs-operator":
+				resources := datafoundation.GetResourceRequirements("ocs-operator")
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+					isChanged = true
+				}
+			case "rook-ceph-operator":
+				resources := datafoundation.GetResourceRequirements("rook-ceph-operator")
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+					isChanged = true
+				}
+			case "ocs-metrics-exporter":
+				resources := datafoundation.GetResourceRequirements("ocs-metrics-exporter")
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+					isChanged = true
+				}
+			default:
+				r.Log.V(-1).Info("Could not find resource requirement", "Resource", container.Name)
+			}
+		}
+	}
+	if isChanged {
+		if err := r.update(csv); err != nil {
+			return fmt.Errorf("failed to update OCS CSV: %v", err)
+		}
+	}
+	return nil
```

Comment:
Your util of `addOwnerAnnotation` does not capture if a change was or wasn't made. As such you might have added an annotation but didn't change anything about resources then the `isChanged` flag will be false and you will not persist your annotation change. That will be a bug

Reply:
Updated